torch/_tensor_str.py

```diff
@@ -80,8 +81,18 @@ def __init__(self, tensor):
             tensor_view = tensor.reshape(-1)
 
         if not self.floating_dtype:
+            self.complex_with_decimal = False
```

Comment:
I think we should change it to `is_non_zero_decimal_val` and perhaps add a comment that it's only used for complex